ci-operator/step-registry/cucushift/upgrade/tolatest/cucushift-upgrade-tolatest-commands.sh

```diff
@@ -0,0 +1,681 @@
+    if [[ ${out} != *"ack-4.${SOURCE_MINOR_VERSION}"* ]]; then
+        echo "Admin ack not required: ${out}" && return
+    fi
+
+    echo "Require admin ack:\n ${out}"
+    local wait_time_loop_var=0 ack_data
+
+    ack_data="$(echo "${out}" | jq -r "keys[]")"
+    for ack in ${ack_data};
+    do
+        # e.g.: ack-4.12-kube-1.26-api-removals-in-4.13
+        if [[ "${ack}" == *"ack-4.${SOURCE_MINOR_VERSION}"* ]]
+        then
+            echo "Admin ack patch data is: ${ack}"
+            oc -n openshift-config patch configmap admin-acks --patch '{"data":{"'"${ack}"'": "true"}}' --type=merge
+        fi
+    done
+    echo "Admin-acks patch gets started"
+
+    echo -e "sleep 5 min wait admin-acks patch to be valid...\n"
+    while (( wait_time_loop_var < 5 )); do
+        sleep 1m
+        echo -e "wait_time_passed=${wait_time_loop_var} min.\n"
+        if ! oc adm upgrade | grep "AdminAckRequired"; then
+            echo -e "Admin-acks patch PASSED\n"
+            return 0
+        else
+            echo -e "Admin-acks patch still in processing, waiting...\n"
+        fi
+        (( wait_time_loop_var += 1 ))
+    done
+    if (( wait_time_loop_var >= 5 )); then
+        echo >&2 "Timed out waiting for admin-acks completing, exiting" && return 1
+    fi
+}
+
+# check if any of cases is enabled via ENABLE_OTA_TEST
+function check_ota_case_enabled() {
+    local case_id
+    local cases_array=("$@")
+    for case_id in "${cases_array[@]}"; do
+        # shellcheck disable=SC2076
+        if [[ " ${ENABLE_OTA_TEST} " =~ " ${case_id} " ]]; then
+            echo "${case_id} is enabled via ENABLE_OTA_TEST on this job."
+            return 0
+        fi
+    done
+    return 1
+}
+
+function set_channel(){
+    x_ver=$( echo "${DUMMY_TARGET_VERSION}" | cut -f1 -d. )
+    y_ver=$( echo "${DUMMY_TARGET_VERSION}" | cut -f2 -d. )
+    ver="${x_ver}.${y_ver}"
+    if ! oc adm upgrade channel candidate-${ver}; then
+        echo "Fail to change channel to candidate-${ver}!"
+        return 1
+    fi
+    local retry=3
+    while (( retry > 0 ));do
+        recommends=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates')
+        if [[ "${recommends}" == "null" ]]; then
+            (( retry -= 1 ))
+            sleep 60
+            echo "No recommended update available! Retry..."
+        else
+            echo "Recommencded update: ${recommends}"
+            break
+        fi
+    done
+}
+function upgrade_tolatest(){
+    echo "Upgrade cluster to the latest recommended update with --to-latest"
+    recommends=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates')
+    if [[ "${recommends}" == "null" ]]; then
+        echo "No recommended update available!"
+        exit 1
+    fi
+    tmp_log=$(mktemp)
+    run_command "oc adm upgrade --to-latest 2>&1 | tee ${tmp_log}"
+    TARGET_VERSION=$(cat "${tmp_log}"|awk '{print $NF}')
```

Comment:
Is it possible to get the `TARGET_VERSION` after `set_channel` but without running `admin_ack` and `cco_annotation`?

Reply:
we need run --to-latest, then it will return the target version, so sounds not.

Reply:
seems not dry-run option to trigger a dummy --to-latest in advance.
```
# ./oc adm upgrade --to-latest --help|grep -E "dry|dry-run|dryrun"
#
```

Reply:
This issue should be able to fix in https://github.com/openshift/release/pull/49743#discussion_r1526174170